assets/js/pages/Users/UsersPage.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { listUsers, deleteUser } from '@lib/api/users';
+import { format, parseISO } from 'date-fns';
+
+import { toast } from 'react-hot-toast';
+
+import Users from './Users';
+
+function UsersPage() {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [deleteUserId, setDeleteUserId] = useState(null);
+  const [userUpdateTrigger, setUserUpdateTrigger] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [userData, setUserData] = useState([]);
+  const [error, setError] = useState(null);
+
+  const navigate = useNavigate();
+
+  const fetchUsers = () => {
+    setLoading(true);
+    listUsers()
+      .then((response) => {
+        setUserData(response.data);
+        setLoading(false);
+      })
+      .catch((_error) => {
+        setError('An error occurred during loading users');
+        setUserData([]);
+        setLoading(false);
+      });
+  };
+
+  const handleDeleteUser = (userId) => {
+    deleteUser(userId)
+      .then(() => {
+        setUserUpdateTrigger(true);
+      })
+      .catch((_error) => {
+        setError('An error occurred during deleting user');
+      });
+  };
+
+  useEffect(() => {
+    fetchUsers();
+    setUserUpdateTrigger(false);
+  }, [userUpdateTrigger]);
```

Comment:
What's this user update trigger?

What we want to trigger? Simply fetch the users initiall on mount, and then , call explicitely the fetch users function to refetch.

Please don't use use effects like that. We don't have any trigger, effects react when things changes and they are meaningful for the domain.